service.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2019 RedHat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"strconv"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	"github.com/prometheus/client_golang/prometheus"
+	v1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	defaultMetricsPath = "/customMetrics"
+	defaultMetricsPort = "8089"
+)
+
+type convertmetrics func()
+
+type PrometheusUser struct {
+	Metricpath            string
+	Metricport            string
+	MetricsList           []prometheus.Collector
+	RecordMetricsFunction convertmetrics
+}
+
+type prometheusInstance struct {
+	metricpath            string
+	metricport            string
+	metriclist            []prometheus.Collector
+	recordMetricsFunction convertmetrics
+}
+
+func WithPath(inputPath string) string {
+	if inputPath == "" {
+		return defaultMetricsPath
+	}
+	return inputPath
+}
+
+func WithPort(inputPort string) string {
+	if inputPort == "" {
+		return defaultMetricsPort
+	}
+	return inputPort
+
+}
+
+var log = logf.Log.WithName("metrics")
```

Comment:
I think this should default to "usermetrics" and could also be configurable.

Reply:
Why is this defined between functions?